docs/src/model_components/carbon-chemistry.md

```diff
@@ -0,0 +1,350 @@
+# [Carbon Chemistry](@id carbon-chemistry)
+
+Our carbon chemistry model follows the best practice described by [dickson2007](@citet), as implemented by e.g. `cbsyst` [branson2023](@citep) and `CO2SYS` [humphreys2022](@citep) in Python.
+
+The carbon chemistry model is primarily used for diagnosing the partial pressure of carbon dioxide in the surface water for [gas exchange with the air](@ref air-sea-gas), but is capable of diagnosing other species such as carbonate concentration, useful in the calculation of calcite dissolution.
+The model works by computing the pH from the total dissolved inorganic carbon and total alkalinity (and optionally silicate, phosphate, boron, sulfate, and fluoride concentration), along with temperature and salinity, which can then be used to find the concentration of different species.
+We will first describe how to use the model, followed by the underlying chemistry and parameterisations.
+
+## Using the model
+To use the carbon chemistry model it is constructed by simply writing:
+```@example carbon-chem
+using OceanBioME
+
+carbon_chemistry = CarbonChemistry()
+```
+
+### Computing ``fCO_2`` and ``pH``
+To compute the fugacity of carbon dioxide (``fCO_2``) you call the model with the `DIC`, `Alk`alinity, `T`emperature, and `S`alinity:
+```@example carbon-chem
+carbon_chemistry(2145, 2448, 25.4, 36.45)
+```
+This is sufficient when computing ``fCO_2`` at the surface, but if we wanted to know ``fCO_2`` at depth where there is higher pressure we can specify the pressure in bar like:
+```@example carbon-chem
+carbon_chemistry(2145, 2448, 25.4, 36.45; P = 100)
+```
+
+We may also be interested in the pH so we can request that be outputted:
```

Comment:
Is it possible to output pCO2 and pH, or would the function need to be called twice?

Reply:
We could do both but I'm not sure what the cleanest way to-do it is. 

I don't want to put any conditions like if X return Y elseif Z return A because the function has to run in a kernel and that can be slower on GPU (although if a kernel is doing this the branch will be the same for all of the threads so this might not actually cause a branching slowdown).

It could be:
```
    return ifelse(return_pCO₂ & return_pH, 
                  (pCO₂, pH),
                  ifele(return_pH, 
                        pH, 
                        ifelse(return_pCO₂,
                               pCO₂,
                               NaN)))
```

but that's kind of messy?

I do see the value of returning them both for diagnostics, but I don't think we'd ever want to use it for the prognostics because if I do e.g.
```
pH = carbon_chemistry(; DIC, Alk, T, S, return_pH=true)
pco2 = carbon_chemistry(; DIC, T, S, pH)
```
the value of `pco2` is different that if I do `carbon_chemistry(; DIC, Alk, T, S)` from the accuracy lost in logging and 10^ing the pH.

So to summarise, I think it would be useful diagnostically, but I can't think of a useful instance for it inside the models where the advantage of not having to call it twice would be the most important.